### DISCUSSION TITLE
Showing Progress?

### DISCUSSION THREAD

**Discussion by Adub648** [SethClydesdale/genki-study-resources#90] [Ideas]
Hi, 

I'm not sure if I am missing something, but is there a way to track your progress on this? If not, this would be a great feature to add, so that people can go through and tick off all of the exercises that they have done.

Thanks, 
Alex Wilson.

**Comment by SethClydesdale** [comment]
Hello @Adub648,

Currently there's no way to track progress. Since it's a static website designed to work online or offline, the only data stored is to cookies/localStorage. So there is a chance one could lose their progress accidentally, should they choose to clear these.

I do like the idea though, and can think of some possible workarounds, such as exporting your progress as a JSON file which you could then re-import into the site to restore your progress (similar to the Custom Study Tools). As for showing the progress, possibly displaying a simple checkmark icon next to the exercise name should be sufficient, with an additional page/mode where you can manually check off exercises that you've completed (in the event of progress loss).

**Reply by yoiang** [reply]
I love this idea as well as how @cocowash is tracking their progress below. Especially as steps working towards being able to generate automatic mixed up vocabulary flashcards or other format reviews

**Comment by cocowash** [comment]
@Adub648, 
If someone is looking for ways to keep progress, I share my method for this. 
I use any add-on that let me highlight a page [Right now i'm using highlighter + notes on mozilla] 
Assign colors to information that you want to remember about a lesson is crucial. 
I use green for current unseen lesson. Red for flunked lessons, and I'm thinking of using blue for useful lessons to keep reviewing regardless of my note. 
By the way, @SethClydesdale, thanks for the hard work, I find this resource pretty useful to review Genki.

**Reply by yoiang** [reply]
@cocowash does the add-on re-fill in the fields you've already worked on as well or are you using it just to mark progress?

**Reply by cocowash** [reply]
> @cocowash does the add-on re-fill in the fields you've already worked on as well or are you using it just to mark progress?
Just to mark progress. In a sense, this methodology works best for overall lessons than for specific answers on those lessons.
